From 51d9e3bcb9fe99a67059e08d7b6fb6ca6a2b75c2 Mon Sep 17 00:00:00 2001
From: Natanael Copa <ncopa@alpinelinux.org>
Date: Thu, 7 Nov 2019 20:41:27 +0000
Subject: [PATCH] Revert "abuild: make built package reproducible"

The introduction of the --pax-options seems to confuse apk and resulted
in `BAD archive' errors.

This reverts commit f04a2ee34b28a38c4349ef1f94686a07afce730f.
---
 abuild.in | 14 +++-----------
 1 file changed, 3 insertions(+), 11 deletions(-)

diff --git a/abuild.in b/abuild.in
index ad9f92b..91f9e1d 100644
--- a/abuild.in
+++ b/abuild.in
@@ -1565,11 +1565,7 @@ create_apks() {
 		# normalize timestamps
 		find . -exec touch -h -d "@$SOURCE_DATE_EPOCH" {} +
 
-		tar --xattrs \
-			--format=posix \
-			--pax-option=exthdr.name=%d/PaxHeaders/%f,delete=atime,delete=ctime,delete=mtime \
-			--mtime="@${SOURCE_DATE_EPOCH}" \
-			-f - -c "$@" | abuild-tar --hash | $gzip -n -9 >"$dir"/data.tar.gz
+		tar --xattrs -f - -c "$@" | abuild-tar --hash | $gzip -9 >"$dir"/data.tar.gz
 
 		msg "Create checksum..."
 		# append the hash for data.tar.gz
@@ -1579,12 +1575,8 @@ create_apks() {
 
 		# control.tar.gz
 		cd "$dir"
-		tar \
-			--format=posix \
-			--pax-option=exthdr.name=%d/PaxHeaders/%f,delete=atime,delete=ctime,delete=mtime \
-			--mtime="@${SOURCE_DATE_EPOCH}" \
-			-f - -c $(cat "$dir"/.metafiles) | abuild-tar --cut \
-			| $gzip -n -9 > control.tar.gz
+		tar -f - -c $(cat "$dir"/.metafiles) | abuild-tar --cut \
+			| $gzip -9 > control.tar.gz
 		abuild-sign -q control.tar.gz || exit 1
 
 		msg "Create $apk"
-- 
2.23.0

